mat-fhir-services/src/main/java/gov/cms/mat/fhir/services/translate/MeasureTranslator.java

```diff
@@ -167,6 +177,41 @@ public Measure translateToFhir(String measureId) {
         return result;
     }
 
+    private void processExtension(Measure fhirMeasure) {
+        fhirMeasure.setExtension(new ArrayList<>());
+        fhirMeasure.getExtension().add(new Extension(EXTENSION_POPULATION_BASIS, new CodeType("boolean")));
+        fhirMeasure.getExtension().add(new Extension(EXTENSION_SOFTWARE_SYSTEM, new Reference("#cqf-tooling")));
+    }
+
+    private void processContained(Measure fhirMeasure) {
+        Device device = new Device();
+        Meta meta = new Meta();
+        meta.setProfile(Collections.singletonList(new CanonicalType("http://hl7.org/fhir/us/cqfmeasures/StructureDefinition/device-softwaresystem-cqfm")));
+        device.setMeta(meta);
+        device.setId("cqf-tooling");
+
+        Device.DeviceVersionComponent deviceVersionComponent = new Device.DeviceVersionComponent();
+        deviceVersionComponent.setValue("1.1.0-SNAPSHOT");
+        device.setVersion(Collections.singletonList(deviceVersionComponent));
+
+        CodeableConcept codeableConcept = new CodeableConcept();
+        Coding coding = new Coding();
+        coding.setSystem("http://hl7.org/fhir/us/cqfmeasures/CodeSystem/software-system-type");
+        coding.setCode("tooling");
+        codeableConcept.setCoding(Collections.singletonList(coding));
+
+        device.setType(codeableConcept);
+
+        fhirMeasure.setContained(Collections.singletonList(device));
+    }
+
+    private void processImprovementNotation(Measure fhirMeasure) {
+        //Todo missing Improvement Notation from matMeasure db
+        CodeableConcept improvementNotation = buildCodeableConcept("FIX ME",
```

Comment:
Are we ok with FIX ME codable concept?

Reply:
No definitely not. This is something I couldn’t figure out on How to get Codeable Concept value for improvement Notation.

Reply:
For now don't include improvementNotation. We need a mat change for it Kathy knows about it.